### PR TITLE
docs(examples): Create description for each crate in examples/*

### LINKED ISSUE #2845: Create description for each crate in examples/*

### PR DESCRIPTION
**Pull request by mqxf**
Resolves #2845.

# Notes:
- Skipping demo-constructor until #3487 is merged due to update to functionality.
- Skipping signal-wait due to #3525.
- I am also fixing missing copyright at the top of files in `examples/*`.

@reviewer-or-team

### CODE REVIEW

**Review by clearloop — COMMENTED** [review]
as we now have documents for examples, we can build docs for them as well https://github.com/gear-tech/gear/blob/master/Makefile#L246

append `--examples`

File: examples/async-custom-entry/src/wasm.rs [hunk 1]
```diff
@@ -16,6 +16,20 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+//! This program demonstrates the use of async `init` entry point, as well as custom entry point
```
Comment (SabaunT) [inline examples/async-custom-entry/src/wasm.rs:19]: I would add some information about the execution. Docs should give some more description on async execution flow in this example.

File: examples/async-init/src/wasm.rs [hunk 2]
```diff
@@ -16,15 +16,17 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! The program demonstrates asynchronous execution and
-//! how to use macros `gstd::async_init`/`gstd::async_main`.
+//! This program demonstrates asynchronous execution and how to use macros `gstd::async_init` and
```
Comment (SabaunT) [inline examples/async-init/src/wasm.rs:19]: please add a bit more info on the role of async here, as from description it seems same stuff can be done without async.

File: examples/async-recursion/src/wasm.rs [hunk 3]
```diff
@@ -16,6 +16,15 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+//! This program demonstrates the use of an async recursive function using the [`async_recursion`]
```
Comment (SabaunT) [inline examples/async-recursion/src/wasm.rs:19]: Give more higher level explanations of the async execution flow.

File: examples/async-tester/src/wasm.rs [hunk 4]
```diff
@@ -1,3 +1,59 @@
...
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! This program demonstrates the use of [`msg`] syscalls from an async environment.
```
Comment (SabaunT) [inline examples/async-tester/src/wasm.rs:19]:
```suggestion
//! This program demonstrates the use of [`gstd::msg`] syscalls from an async environment.
```

File: examples/async-tester/src/wasm.rs [hunk 5]
```diff
@@ -1,3 +1,59 @@
...
+//! [`SendBytesWithGas(gas)`] uses the [`send_bytes_with_gas_for_reply()`] syscall to send a message
+//! back to the source, containing the `kind` encoded as bytes in the payload and `gas` as the gas
+//! limit.
+//!
+//! [`SendCommit`] uses the [`MessageHandle`], pushing the `kind` encoded as bytes and using
+//! [`commit_for_reply()`] to send the message back to the source.
+//!
+//! [`SendCommitWithGas(gas)`] uses the [`MessageHandle`], pushing the `kind` encoded as bytes and
+//! using [`commit_with_gas_for_reply()`] to send the message back to the source with `gas` as the
+//! gas limit.
```
Comment (SabaunT) [inline examples/async-tester/src/wasm.rs:42]: You don't need to literally explain every step of the program. These steps are clear, btw. State main of this tester crate (look at the usage).

File: examples/async/src/wasm.rs [hunk 6]
```diff
@@ -16,6 +16,27 @@
...
+//! `Handle` is async and gets a [`Command`] in the payload, executes a certain action based on it,
+//! and send a [`reply()`] with a payload containing the id of the current message. There are two commands
+//! that can be executed: [`Common`] and [`Mutex`].
+//!
+//! [`Common`] sends three async messages to the ping program, with the payload "PING",
+//! awaits the reply and asserts that the reply is "PONG".
+//!
+//! [`Mutex`] asynchronously locks the mutex, awaiting it, sends a message back to the
+//! source of the current message, containing the current message id in the payload, and then
+//! it sends a ping message, awaiting the reply and asserting that the reply is "PONG".
```
Comment (SabaunT) [inline examples/async/src/wasm.rs:34]: it's not clear what this wasms does, what's the purpose of that.

**Review by SabaunT — COMMENTED** [review]
I believe the approach to docs should be different. Instead of writing what literally every call in each function, you'd better give first some high-level, conceptual comments on purpose of this particular example, execution flow and how it reaches the purpose of that wasm. Entry points details about each procedure in them can be reduced.

Also to be clear, this is a general comment for all documented examples, not only "async" ones.

**Comment by breathx** [comment]
Please re-open once work continued

**Pull request closed without merging**